dali/test/python/test_operator_readers_webdataset_requirements.py

```diff
@@ -0,0 +1,427 @@
+
+    extract_dir = generate_temp_extract(tar_file_path)
+    equivalent_files = sorted(
+        glob(extract_dir.name + "/*"), key=lambda s: int(s[s.rfind("/") + 1 : s.rfind(".")])
+    )[:num_samples]
+
+    wds_pipeline = webdataset_raw_pipeline(
+        tar_file_path,
+        index_file.name,
+        ["jpg", "cls"],
+        missing_component_behavior="error",
+        batch_size=test_batch_size,
+        device_id=0,
+        num_threads=1,
+    )
+    assert_raises(RuntimeError, wds_pipeline.build)
+
+
+def test_different_components():
+    global test_batch_size
+    num_samples = 1000
+    tar_file_path = get_dali_extra_path() + "/db/webdataset/MNIST/scrambled.tar"
+    index_file = generate_temp_index_file(tar_file_path)
+
+    extract_dir = generate_temp_extract(tar_file_path)
+    equivalent_files = sorted(
+        glob(extract_dir.name + "/*"), key=lambda s: int(s[s.rfind("/") + 1 : s.rfind(".")])
+    )
+
+    compare_pipelines(
+        webdataset_raw_pipeline(
+            tar_file_path,
+            index_file.name,
+            ["jpg", "txt;cls"],
+            missing_component_behavior="empty",
+            batch_size=test_batch_size,
+            device_id=0,
+            num_threads=1,
+        ),
+        file_reader_pipeline(
+            equivalent_files,
+            ["jpg", {"txt", "cls"}],
+            batch_size=test_batch_size,
+            device_id=0,
+            num_threads=1,
+        ),
+        test_batch_size,
+        math.ceil(num_samples / test_batch_size),
+    )
+
+
+def test_dtypes():
+    global test_batch_size
+    num_samples = 100
+    tar_file_path = get_dali_extra_path() + "/db/webdataset/sample-tar/dtypes.tar"
+    index_file = generate_temp_index_file(tar_file_path)
+
+    wds_pipeline = webdataset_raw_pipeline(
+        tar_file_path,
+        index_file.name,
+        ["float16", "int32", "float64"],
+        dtypes=[
+            dali.types.FLOAT16,
+            dali.types.INT32,
+            dali.types.FLOAT64
+        ],
+        batch_size=test_batch_size,
+        device_id=0,
+        num_threads=1,
+    )
+    wds_pipeline.build()
+    for sample_idx in range(num_samples):
+        if sample_idx % test_batch_size == 0:
+            f16, i32, f64 = wds_pipeline.run()
+        assert (f16.as_array()[sample_idx % test_batch_size] == [float(sample_idx)] * 10).all()
+        assert (i32.as_array()[sample_idx % test_batch_size] == [int(sample_idx)] * 10).all()
+        assert (f64.as_array()[sample_idx % test_batch_size] == [float(sample_idx)] * 10).all()
+
+
+def test_wds_sharding():
+    global test_batch_size
+    num_samples = 3000
+    tar_file_paths = [
+        get_dali_extra_path() + "/db/webdataset/MNIST/devel-0.tar",
+        get_dali_extra_path() + "/db/webdataset/MNIST/devel-1.tar",
+        get_dali_extra_path() + "/db/webdataset/MNIST/devel-2.tar",
+    ]
+    index_files = [generate_temp_index_file(tar_file_path) for tar_file_path in tar_file_paths]
```

Comment:
I see repeating pattern here:
```
tar_file_paths = (...)
index_files = (...)
extract_dirs = (...)
equivalent_files = (...)
compare_pipelines(
        webdataset_raw_pipeline(...)
        file_reader_pipeline(...)
   (...)
)
```
Maybe it can be enclosed in a function and test args could be separated test cases in a list or something?

Reply:
the thing is that different things are being changed here and that would mean a lot of arguments. I was actually thinking about it, maybe I will try a bit harder to compact it